src/app/space/create/deployments/deployments-donut/deployments-donut.component.spec.ts

```diff
@@ -50,60 +48,57 @@ describe('DeploymentsDonutComponent', () => {
+  initContext(DeploymentsDonutComponent, HostComponent,
+    {
+      declarations: [FakeDeploymentsDonutChartComponent],
       providers: [{ provide: DeploymentsService, useFactory: () => mockSvc }]
+    },
+    component => {
+      component.mini = false;
+      component.spaceId = 'space';
+      component.applicationId = 'application';
+      component.environment = { name: 'environmentName' } as Environment;
     });
 
-    fixture = TestBed.createComponent(DeploymentsDonutComponent);
-    component = fixture.componentInstance;
-
-    component.mini = false;
-    component.spaceId = 'space';
-    component.applicationId = 'application';
-    component.environment = { name: 'environmentName' } as Environment;
-
-    fixture.detectChanges();
-  });
-
-  it('should use pods data for initial desired replicas', () => {
-    expect(component.desiredReplicas).toEqual(2);
+  it('should use pods data for initial desired replicas', function (this: Context) {
+    expect(this.testedDirective.desiredReplicas).toEqual(2);
   });
 
-  it('should increment desired replicas on scale up by one', () => {
+  it('should increment desired replicas on scale up by one', function (this: Context) {
     let desired = 2;
-    expect(component.desiredReplicas).toBe(desired);
+    expect(this.testedDirective.desiredReplicas).toBe(desired);
 
-    component.scaleUp();
-    fixture.detectChanges();
-    expect(component.desiredReplicas).toBe(desired + 1);
+    this.testedDirective.scaleUp();
+    this.detectChanges();
+    expect(this.testedDirective.desiredReplicas).toBe(desired + 1);
   });
 
-  it('should decrement desired replicas on scale down by one', () => {
+  it('should decrement desired replicas on scale down by one', function (this: Context) {
     let desired = 2;
-    expect(component.desiredReplicas).toBe(desired);
+    expect(this.testedDirective.desiredReplicas).toBe(desired);
 
-    component.scaleDown();
-    fixture.detectChanges();
-    expect(component.desiredReplicas).toBe(desired - 1);
+    this.testedDirective.scaleDown();
+    this.detectChanges();
+    expect(this.testedDirective.desiredReplicas).toBe(desired - 1);
   });
 
-  it('should not decrement desired replicas below zero when scaling down', () => {
+  it('should not decrement desired replicas below zero when scaling down', function (this: Context) {
     let desired = 2;
-    expect(component.desiredReplicas).toBe(desired);
+    expect(this.testedDirective.desiredReplicas).toBe(desired);
 
-    component.scaleDown();
-    fixture.detectChanges();
-    expect(component.desiredReplicas).toBe(desired - 1);
+    this.testedDirective.scaleDown();
+    this.detectChanges();
+    expect(this.testedDirective.desiredReplicas).toBe(desired - 1);
 
-    component.scaleDown();
-    fixture.detectChanges();
-    expect(component.desiredReplicas).toBe(0);
+    this.testedDirective.scaleDown();
+    this.detectChanges();
+    expect(this.testedDirective.desiredReplicas).toBe(0);
```

Comment:
Oops. I think I forgot another scaleDown to check that when it's 0, another scaleDown call doesn't make it -1 (should not decrement below zero). Could you add that while you're here?